### PR TITLE
CDAR-732: Remove redundant requests

### PR DESCRIPTION
**Pull request by paulbourelly999**
# PR Details
## Description
Remove redundant requests to XMLRPC CARLA CDA Sim adapter server. Also removed redundant calls to requestNextTime

## Related Jira Key
[CDDAR-732
](https://usdot-carma.atlassian.net/browse/CDAR-732)

## Motivation and Context
Improve performance and clean up logs

## How Has This Been Tested?
Tested in CDASim deployment

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: co-simulation/fed/mosaic-carla/src/main/java/org/eclipse/mosaic/fed/carla/ambassador/CarlaAmbassador.java [hunk 1]
```diff
@@ -360,26 +356,27 @@ public synchronized void processTimeAdvanceGrant(long time) throws InternalFeder
...
+                for (DetectedObjectInteraction detectionInteraction: detectedObjectInteractions) {
+                    this.rti.triggerInteraction(detectionInteraction);
+                }
                 nextTimeStep += carlaConfig.updateInterval * TIME.MILLI_SECOND;
                 isSimulationStep = false;
+                rti.requestAdvanceTime(nextTimeStep , 0, (byte) 2);
             }
             // TODO: What is this. Why are we request a time advance based on this counter and 
             // what is it counting. It is labeled as counting the times we attempt to connect to 
             // CARLA but it seems to increment every time processTimeAdvanceGrant is called
```
Comment (MishkaMN) [inline co-simulation/fed/mosaic-carla/src/main/java/org/eclipse/mosaic/fed/carla/ambassador/CarlaAmbassador.java:377]: we should remove this comment now

**Review by MishkaMN — CHANGES_REQUESTED** [review]

**Comment by kjrush** [comment]
Seems good to me, I'm okay with approving it just on a review, but want to confirm the testing. How long were the test cases you ran? I think we were seeing the previous livelock issue occurring at like 5ish minutes into simulation execution. To my recollection it was stochastic though, it just happened that about 5 minutes was enough timesteps that it was pretty probable to occur.

If we can confirm that's not a problem, I'm down to approve and merge.

**Comment by paulbourelly999** [comment]
> Seems good to me, I'm okay with approving it just on a review, but want to confirm the testing. How long were the test cases you ran? I think we were seeing the previous livelock issue occurring at like 5ish minutes into simulation execution. To my recollection it was stochastic though, it just happened that about 5 minutes was enough timesteps that it was pretty probable to occur.
> 
> If we can confirm that's not a problem, I'm down to approve and merge.

Hey @kjrush out test case is significantly shorted than 5 minutes. Regardless if this bug is there are not after 5 minutes, this is definitely not the fix and seems to at least partially negatively impact performance. I will try to run a 10 minute simulation and see what happens

File: co-simulation/fed/mosaic-carla/src/main/java/org/eclipse/mosaic/fed/carla/ambassador/CarlaAmbassador.java (hunk 1 above)
Comment (paulbourelly999) [reply co-simulation/fed/mosaic-carla/src/main/java/org/eclipse/mosaic/fed/carla/ambassador/CarlaAmbassador.java]: addressed

**Review by MishkaMN — APPROVED** [review]

**Comment by paulbourelly999** [comment]
> > Seems good to me, I'm okay with approving it just on a review, but want to confirm the testing. How long were the test cases you ran? I think we were seeing the previous livelock issue occurring at like 5ish minutes into simulation execution. To my recollection it was stochastic though, it just happened that about 5 minutes was enough timesteps that it was pretty probable to occur.
> > If we can confirm that's not a problem, I'm down to approve and merge.
> 
> Hey @kjrush out test case is significantly shorted than 5 minutes. Regardless if this bug is there are not after 5 minutes, this is definitely not the fix and seems to at least partially negatively impact performance. I will try to run a 10 minute simulation and see what happens

@kjrush Tested for 5 minute simulation, other services crashed but no CDASim

**Review by MishkaMN — APPROVED** [review]

**Review by MishkaMN — APPROVED** [review]

**Comment by paulbourelly999** [comment]
I am going to merge this and independently see I the issues with RTI event priority dead lock can be recreated. It is possible this change re introduces this behavior but with VRU UC simulation scenarios this behavior seems to not be present. Might be more likely with the addition of SUMO actors. The fix that is undone by this PR is likely not the long term fix. If this PR re introduces the bug we can make a decision independently about whether to find a more long term fix or revert to this fix

**Pull request merged**